satellite/metainfo/metainfo.go

```diff
@@ -523,12 +524,11 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 			return Error.New("all pieces needs to have the same size")
 		}
 
-		// we repair when the number of healthy files is less than or equal to the repair threshold
-		// except for the case when the repair and success thresholds are the same (a case usually seen during testing)
-		if int32(len(remotePieces)) <= remote.Redundancy.RepairThreshold && int32(len(remotePieces)) < remote.Redundancy.SuccessThreshold {
-			return Error.New("Number of valid pieces (%d) is less than or equal to the repair threshold (%d)",
+		// todo: is it okay that this is using endpoint.requiredRSConfig instead of pointer rs values?
+		if len(remotePieces) < endpoint.requiredRSConfig.SuccessThreshold {
```

Comment:
I think we want to make this check against the actual pointer.

Reply:
why the actual pointer and not the endpoint settings?

Reply:
Because we want to check that the uplink reached the success threshold for the specific object it is committing.

It's true the satellite currently checks that redundancy scheme of the pointer matches exact values, but there is a plan to change this in future with a more relaxed check - that the pointer's redundancy falls in some ranges.

Reply:
ah okay. do you know if I should leave this whole `and` statement: 
```if int32(len(remotePieces)) <= remote.Redundancy.RepairThreshold && int32(len(remotePieces)) < remote.Redundancy.SuccessThreshold {
```
or just the latter half?